tdmq/client/client.py

```diff
@@ -160,19 +164,26 @@ def get_timeseries(self, code, args):
         _logger.debug('get_timeseries(%s, %s)', code, args)
         return self._do_get(f'sources/{code}/timeseries', params=args)
 
-    @requires_connection
-    def save_tiledb_frame(self, tdmq_id, slot, data):
+
+    def open_array(self, tdmq_id, mode='r'):
         aname = self._source_data_path(tdmq_id)
-        with tiledb.DenseArray(aname, mode='w', ctx=self.tiledb_ctx) as A:
-            A[slot:slot + 1] = data
+        return tiledb.open(aname, mode=mode, ctx=self.tiledb_ctx)
+
+
+    def close_array(self, tiledb_array):
+        tiledb_array.close()
+
 
     @requires_connection
-    def fetch_non_scalar_slice(self, tdmq_id, tiledb_indices, args):
-        # FIXME hwired on tiledb
+    def save_tiledb_frame(self, tiledb_array, slot, data):
```

Comment:
is this change backwards-compatible? it seems the signature is different (tdmq_id vs tiledb_array)

Reply:
those methods are different, you're right. I should have said that the changes are backwards compatible as long as the client code uses the Source and TimeSeries classes to insert and retrieve data.  The Client interface indeed changes.

Reply:
ok